org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/diagnostics/LSPDiagnosticsToMarkers.java

```diff
@@ -207,7 +207,7 @@ private IMarker getExistingMarkerFor(IDocument document, Diagnostic diagnostic,
 		for (IMarker marker : remainingMarkers) {
 			try {
 				if (LSPEclipseUtils.toOffset(diagnostic.getRange().getStart(), document) == MarkerUtilities.getCharStart(marker)
-						&& LSPEclipseUtils.toOffset(diagnostic.getRange().getEnd(), document) == MarkerUtilities.getCharEnd(marker)
+						&& (LSPEclipseUtils.toOffset(diagnostic.getRange().getEnd(), document) == MarkerUtilities.getCharEnd(marker) || Objects.equals(diagnostic.getRange().getStart(), diagnostic.getRange().getEnd()))
```

Comment:
Since when we create the marker, we already set the end position to be one char forward, would the existing condition not match already? Why do we need a new one?

Reply:
> Since when we create the marker, we already set the end position to be one char forward

Only in case of empty range

> would the existing condition not match already? 

No, it does not match this case.

Reply:
thanks